crates/hdk3/src/lib.rs

```diff
@@ -1,9 +1,482 @@
 pub mod x_salsa20_poly1305;
+
+/// Rexporting the paste macro as it is used internally and may help structure downstream code.
 pub use paste;
+
+/// Tools to interrogate source chains.
+///
+/// Interacting with a source chain is very different to the DHT.
+///
+/// - Source chains have a linear history guaranteed by header hashes
+/// - Source chains have a single owner/author signing every chain element
+/// - Source chains can be iterated over from most recent back to genesis by following the header hashes as references
+/// - Source chains contain interspersed system and application entries
+/// - Source chains contain both private (local only) and public (broadcast to DHT) elements
+///
+/// There is a small DSL provided by `query` that allows for inspecting the current agent's local source chain.
+/// Typically it will be faster, more direct and efficient to query local data than dial out to the network.
+/// It is also possible to query local private entries.
+///
+/// Agent activity for any other agent on the network can be fetched.
+/// The agent activity is _only the headers_ of the remote agent's source chain.
+/// Agent activity allows efficient building of the history of an agent.
+/// Agent activity is retrieved from a dedicated neighbourhood centered around the agent.
+/// The agent's neighbourhood also maintains a passive security net that guards against attempted chain forks and/or rollbacks.
+/// The same query DSL for local chain queries is used to filter remote agent activity headers.
+pub mod chain;
+
+/// Create and verify signatures for serializable Rust structures and raw binary data.
+///
+/// The signatures are always created with the [Ed25519](https://en.wikipedia.org/wiki/EdDSA) algorithm by the secure keystore (lair).
+///
+/// Agent public keys that identify agents are the public half of a signing keypair.
+/// The private half of the signing keypair never leaves the secure keystore and certainly never touches wasm.
+///
+/// If a signature is requested for a public key that has no corresponding private key in lair, the signing will fail.
+///
+/// Signatures can always be verified with the public key alone so can be done remotely (by other agents) and offline, etc.
+///
+/// The elliptic curve used by the signing algorithm is the same as the curve used by the encryption algorithms but is _not_ constant time (because signature verification doesn't need to be).
+///
+/// In general it is __not a good idea to reuse signing keys for encryption__ even if the curve is the same, without mathematically translating the keypair, and even then it's dubious to do so.
+pub mod ed25519;
+
+/// Request contextual information from the holochain host.
+///
+/// The holochain host has additional runtime context that the wasm may find useful and cannot produce for itself including:
+///
+/// - The calling agent
+/// - The current bundle
```

Comment:
No need to change this broadly now, but just a note that we're probably just calling this an "app" rather than a "bundle"

```suggestion
/// - The current app (bundle of DNAs)
```